From aee4e2c8c7ec312b8a0423f08964fa7dcc31611c Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Mon, 12 Feb 2024 12:22:02 +0200
Subject: [PATCH] [SW-137121] gaudi3: erom build: erom start offset

In gaudi3, pif arc binaries are copied to arc1 iccm base
address at zero offset and then copied to pif arc iccm/dccm.
So, Erom pcie code can't start at arc1 iccm base address, we have
added start offset of 0x1000 (4Kb).

dev_ext_rom is directly included in .S instead of adding it in
common.h

erom text length is changed to 64Kb, which is good enough to support
erom recovery (size is 40Kb).

Change-Id: I8e21ec3165fa2cb2e7720fdaded953b078cd5d89
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/app/ext_rom/common.h b/app/ext_rom/common.h
index 92a91fe..7b6e440 100644
--- a/app/ext_rom/common.h
+++ b/app/ext_rom/common.h
@@ -2,5 +2,3 @@
  *
  * Copyright (C) 2021 HabanaLabs Ltd.
  */
-
-#include <dev_ext_rom.h>
diff --git a/app/ext_rom/pcie_erom.lds.S b/app/ext_rom/pcie_erom.lds.S
index 8c43c57..07c8fed 100644
--- a/app/ext_rom/pcie_erom.lds.S
+++ b/app/ext_rom/pcie_erom.lds.S
@@ -4,7 +4,7 @@
  * All Rights Reserved.
  */
 
-#include "common.h"
+#include <dev_ext_rom.h>
 
 MEMORY { .sram : ORIGIN = EXT_ROM_PCIE_TEXT_BASE, LENGTH = EXT_ROM_TEXT_LENGTH }
 OUTPUT_FORMAT("elf32-littlearc", "elf32-bigarc", "elf32-littlearc")
diff --git a/app/ext_rom/recovery_erom.lds.S b/app/ext_rom/recovery_erom.lds.S
index c594559..807ef3b 100644
--- a/app/ext_rom/recovery_erom.lds.S
+++ b/app/ext_rom/recovery_erom.lds.S
@@ -4,7 +4,7 @@
  * All Rights Reserved.
  */
 
-#include "common.h"
+#include <dev_ext_rom.h>
 
 MEMORY { .sram : ORIGIN = EXT_ROM_RECOVERY_TEXT_BASE, LENGTH = EXT_ROM_TEXT_LENGTH }
 OUTPUT_FORMAT("elf32-littlearc", "elf32-bigarc", "elf32-littlearc")
diff --git a/arch/gaudi3/arc/dev_ext_rom.h b/arch/gaudi3/arc/dev_ext_rom.h
index 9724d7f..0972417 100644
--- a/arch/gaudi3/arc/dev_ext_rom.h
+++ b/arch/gaudi3/arc/dev_ext_rom.h
@@ -6,11 +6,12 @@
 #ifndef DEV_EXT_ROM_H_
 #define DEV_EXT_ROM_H_
 
-#define EXT_ROM_START			(0x23040000)
-#define EXT_ROM_TEXT_LENGTH		(128*1024)
+#include <gaudi3/gaudi3_mem_addr_dt.h>
+
+#define EXT_ROM_TEXT_LENGTH		(64*1024)
 /* No header bytes. Hence the text base is same as erom start address */
-#define EXT_ROM_PCIE_TEXT_BASE		(EXT_ROM_START)
-#define EXT_ROM_RECOVERY_TEXT_BASE	(EXT_ROM_START)
-#define EXT_ROM_TPM_TEXT_BASE		(EXT_ROM_START)
+#define EXT_ROM_PCIE_TEXT_BASE		(EROM_START_OFFSET)
+#define EXT_ROM_RECOVERY_TEXT_BASE	(EROM_START_OFFSET)
+#define EXT_ROM_TPM_TEXT_BASE		(EROM_START_OFFSET)
 
 #endif
